src/internal/ButtonBase.spec.js

```diff
@@ -301,15 +309,19 @@ describe('<ButtonBase />', () => {
       const event = new window.Event('keyup');
       event.which = keycode('tab');
       window.dispatchEvent(event);
+    });
+
+    it('should not set keyboard focus before time has passed', () => {
+      assert.strictEqual(wrapper.state('keyboardFocused'), false, 'should not be keyboardFocused');
+    });
+
+    it('should listen for tab presses and set keyboard focus', () => {
+      clock.tick(instance.keyboardFocusCheckTime * instance.keyboardFocusMaxCheckTimes);
+      assert.strictEqual(wrapper.state('keyboardFocused'), true, 'should be keyboardFocused');
+    });
 
-      setTimeout(() => {
-        assert.strictEqual(
-          wrapper.state('keyboardFocused'),
-          true,
-          'should be keyboardFocused',
-        );
-        done();
-      }, 200);
+    after(() => {
+      clock.restore();
```

Comment:
I would rather have the `after` block at the top, right under the `before` bloc. It's our convention. Plus the setup and teardown phases are deeply linked.